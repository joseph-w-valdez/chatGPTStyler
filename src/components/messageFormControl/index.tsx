export {
    MessageFormControl as MessageFormControl,
    MessageFormControlProps,
} from "./component";
