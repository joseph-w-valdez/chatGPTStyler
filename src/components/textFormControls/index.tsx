export { TextFormControls, TextFormControlsProps } from "./component";
