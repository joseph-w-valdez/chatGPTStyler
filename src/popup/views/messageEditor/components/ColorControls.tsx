import { SettingsType } from "@src/lib/utilities/googleStorage";
import { sendMessageToTab } from "@src/shared/utils";
import React from "react";

export interface ColorControlsProps {
    setLiveChanges: React.Dispatch<React.SetStateAction<SettingsType>>;
    liveChanges: SettingsType;
    setIsEditing: React.Dispatch<React.SetStateAction<boolean>>;
}
export type colorSetting = "messageColor" | "textColor";

export function ColorControls({
    setLiveChanges,
    liveChanges,
    setIsEditing,
}: ColorControlsProps): JSX.Element {
    const colorSettings: colorSetting[] = ["messageColor", "textColor"];

    // creates color controls. we are making one for user and chat gpt.
    const mapColorSettings = (userType: string, index: number) => {
        const isUser = userType === "User";
        return (
            <div
                key={index}
                className="flex flex-col justify-center items-center bg-violet-500 rounded-md p-3 gap-1 font-medium w-full"
            >
                <span
                    className=" text-center p-1 w-24 rounded-md"
                    style={{
                        backgroundColor:
                            liveChanges[
                                `${
                                    isUser
                                        ? "messageColorUserStyle"
                                        : "messageColorNonUserStyle"
                                }`
                            ],
                        color: liveChanges[
                            `${
                                isUser
                                    ? "textColorUserStyle"
                                    : "textColorNonUserStyle"
                            }`
                        ],
                    }}
                >
                    {`${userType} Color`}
                </span>
                <div className="grid grid-cols-2 gap-1 w-full text-white">
                    {colorSettings.map((setting, index) =>
                        mapSettingInputs(setting, index, isUser),
                    )}
                    <span className="rounded-md font-medium text-center p-0 m-0">
                        BG
                    </span>
                    <span className="rounded-md font-medium text-center p-0 m-0">
                        Text
                    </span>
                </div>
            </div>
        );
    };
    // creates input elements for color controls. we currently have text color, and bg color
    const mapSettingInputs = (
        setting: colorSetting,
        index: number,
        isUser: boolean,
    ) => {
        const settingsKey: keyof SettingsType = `${setting}${
            isUser ? "User" : "NonUser"
        }Style`;

        const handleOnChange = (
            e: React.ChangeEvent<HTMLInputElement>,
            settingsKey: keyof SettingsType,
        ) => {
            setLiveChanges({
                ...liveChanges,
                [settingsKey]: e.currentTarget.value,
            });
            sendMessageToTab(settingsKey, e.currentTarget.value);
            setIsEditing(true);
        };
        return (
            <input
                key={index}
                className={`text-center w-full rounded-md h-8`}
                type="color"
                id={`${settingsKey}`}
                value={liveChanges[settingsKey]}
                onChange={(e) => handleOnChange(e, settingsKey)}
            />
        );
    };

    return (
        <div className="grid grid-cols-2 gap-2 place-items-center w-full">
            {["User", "Chat-GPT"].map(mapColorSettings)}
        </div>
    );
}
