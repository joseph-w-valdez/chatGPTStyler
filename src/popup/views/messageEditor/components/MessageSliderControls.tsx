import { SettingsType } from "@src/lib/utilities/googleStorage";
import { sendMessageToTab } from "@src/shared/utils";
import React from "react";

export interface SliderControlsProps {
    setLiveChanges: React.Dispatch<React.SetStateAction<SettingsType>>;
    liveChanges: SettingsType;
    setIsEditing: React.Dispatch<React.SetStateAction<boolean>>;
}
export type colorSetting = "messageColor" | "textColor";

export function MessageSliderControls({
    setLiveChanges,
    liveChanges,
    setIsEditing,
}: SliderControlsProps): JSX.Element {
    // using this to create data we need so we can map through and have clean callback
    class InputSetting {
        name: string = "";
        id: keyof SettingsType = "messageMaxWidthStyle";
        valueType: "px" | "%" = "px";

        constructor(
            name: string,
            id: keyof SettingsType = "messageMaxWidthStyle",
            valueType: "px" | "%" = "px",
        ) {
            this.name = name;
            this.id = id;
            this.valueType = valueType;
        }
    }
    const inputSettings = [
        new InputSetting("Message Width", "messageMaxWidthStyle", "%"),
        new InputSetting("Message Padding", "messagePaddingStyle", "px"),
        new InputSetting("Message Border", "messageBorderRadiusStyle", "px"),
        new InputSetting("Input Box Width", "inputBoxMaxWidthStyle", "%"),
    ];

    const mapInputSettings = (setting: InputSetting, index: number) => {
        const bgColors = [
            "rgba(145, 10, 103",
            "rgba(114, 4, 85",
            "rgba(60, 7, 83",
            "rgba(3, 6, 55",
        ];

        const handleOnChange = (
            e: React.ChangeEvent<HTMLInputElement>,
            settingKey: keyof SettingsType,
        ) => {
            setIsEditing(true);
            sendMessageToTab(settingKey, e.currentTarget.value);
            setLiveChanges({
                ...liveChanges,
                [settingKey]: e.currentTarget.value,
            });
        };
        return (
            <div className="flex justify-between items-center" key={index}>
                <label
                    htmlFor={setting.id}
                    className={`basis-1/2 p-2 text-white rounded-md font-medium text-center mr-2`}
                    style={{
                        backgroundColor: `${bgColors[index]})`,
                    }}
                >{`${setting.name}:`}</label>{" "}
                <div className="relative flex ">
                    <input
                        className={`outline-none px-2 py-2 text-right w-full rounded-l-sm`}
                        type="text"
                        maxLength={2}
                        style={{
                            backgroundColor: `${bgColors[index]}, 0.1)`,
                        }}
                        value={liveChanges[setting.id]}
                        id={setting.id}
                        onChange={(e) => handleOnChange(e, setting.id)}
                    ></input>
                    <input
                        type="range"
                        min="1"
                        max="95"
                        value={liveChanges[setting.id]}
                        className="absolute w-full h-0.5 left-0 bottom-0"
                        onChange={(e) => handleOnChange(e, setting.id)}
                        step={"1"}
                        style={{ accentColor: `${bgColors[index]})` }}
                    ></input>
                </div>
                <div
                    className={`py-2 rounded-r-sm`}
                    style={{
                        backgroundColor: `${bgColors[index]}, 0.1)`,
                    }}
                >
                    {setting.valueType}
                </div>
            </div>
        );
    };

    return (
        <div className={`grid grid-cols-1 gap-y-3 `}>
            {inputSettings.map(mapInputSettings)}
        </div>
    );
}
