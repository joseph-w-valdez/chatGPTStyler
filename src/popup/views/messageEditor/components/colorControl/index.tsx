export { ColorControls, ColorControlsProps } from "./component";
