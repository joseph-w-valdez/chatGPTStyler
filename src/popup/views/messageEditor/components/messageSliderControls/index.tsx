export {
    MessageSliderControls,
    SliderControlsProps,
    colorSetting,
} from "./component";
