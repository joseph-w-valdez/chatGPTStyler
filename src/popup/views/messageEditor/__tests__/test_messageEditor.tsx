import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { MessageEditor, MessageEditorProps } from "../index";
import { MessageSliderControls } from "../components/MessageSliderControls";
import { ColorControls } from "../components/ColorControls";

describe("MessageEditor Component", () => {
    let wrapper: ShallowWrapper;
    const mockProps: MessageEditorProps = {
        liveSettings: {
            messageColorUserStyle: "",
            messageColorNonUserStyle: "",
            messageMaxWidthStyle: "",
            messagePaddingStyle: "",
            messageBorderRadiusStyle: "",
            inputBoxMaxWidthStyle: "",
            textColorUserStyle: "",
            textColorNonUserStyle: "",
            textSizeUserStyle: "",
            textSizeNonUserStyle: "",
            textWeightUserStyle: "",
            textWeightNonUserStyle: "",
            messageButtonsVisibilityStyle: true,
        },
        setLiveSettings: jest.fn,
    };

    beforeEach(() => {
        wrapper = shallow(<MessageEditor {...mockProps} />);
    });

    it("renders correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("calls setLiveChanges on user message color change", () => {
        const liveChanges = mockProps.liveSettings;
        wrapper.find(MessageSliderControls).at(0).prop("setLiveChanges")(
            liveChanges,
        );
        expect(mockProps.setLiveSettings).toHaveBeenCalledWith(liveChanges);
    });

    it("calls setLiveChanges on chat message color change", () => {
        const liveChanges = mockProps.liveSettings;
        wrapper.find(ColorControls).at(1).prop("setLiveChanges")(liveChanges);
        expect(mockProps.setLiveSettings).toHaveBeenCalledWith(liveChanges);
    });

    it("calls messageMaxWidthLiveChange on message max width change", () => {
        const event = {
            target: { value: "50%" },
        } as React.ChangeEvent<HTMLInputElement>;
        wrapper.find("#messageMaxWidthStyle").simulate("change", event);
        expect(mockProps.setLiveSettings).toHaveBeenCalledWith(event);
    });
});
