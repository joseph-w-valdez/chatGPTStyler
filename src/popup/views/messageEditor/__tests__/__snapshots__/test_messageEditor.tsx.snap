// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MessageEditor renders correctly 1`] = `
<div class="grid grid-cols-1 gap-y-3 px-3 pb-2 animate-fade-in">
  <MessageSliderControls
    liveChanges={[Object]}
    setLiveChanges={[Function]}
    setIsEditing={[Function]}
  />
  <ColorControls
    liveChanges={[Object]}
    setLiveChanges={[Function]}
    setIsEditing={[Function]}
  />
  <FormButtons
    isEditing={false}
    liveSettings={[Object]}
    savedSettings={[Object]}
    setLiveSettings={[Function]}
    setSavedSettings={[Function]}
    setIsEditing={[Function]}
  />
</div>
`;
